src/compile/scale/type.ts

```diff
@@ -33,8 +36,8 @@ export default function type(
       return defaultScaleType;
     }
 
-    // Check if explicitly specified scale type is supported by the channel
-    if (supportScaleType(channel, specifiedType)) {
+    // Check if explicitly specified scale type is supported by the channel and data type
+    if (supportScaleType(channel, specifiedType) && dataTypeMatchScaleType(specifiedType, fieldDef)) {
```

Comment:
The error below outputs error for scaleTypeNotWorkWithChannel, which can be incorrect.

Better make it throws the right warnings for not work with channel / not work with data type.